File: Week 1/Problem Set 1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int bricks);

int main(void)
{
    // Prompt the user for the pyramid's height
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1);

    // Print a pyramid of that height
    for (int i = 0; i < n; i++)
    {
        // Print row of bricks
        print_row(n, i + 1);
    }
}

void print_row(int spaces, int bricks)
{
    // Print spaces
    for (int i = 0; i < spaces - bricks; i++)
    {
        printf(" ");
    }
    // Print bricks
    for (int i = 0; i < bricks; i++)
    {
        printf("#");
    }
    printf("\n");
}

File: Week 6/Problem Set 6/sentimental-mario-less/mario.py
def print_row(spaces, bricks):
    for i in range(spaces - bricks):
        print(" ", end="")

    for i in range(bricks):
        print("#", end="")

    print()


def main():
    while True:
        try:
            n = int(input("Height: "))
            if 1 <= n <= 8:
                break
        except ValueError:
            continue

    for i in range(n):
        print_row(n, i + 1)


if __name__ == "__main__":
    main()
